### DISCUSSION TITLE
HopskipJump Attack

### DISCUSSION THREAD

**Discussion by fukashi-hatake** [Trusted-AI/adversarial-robustness-toolbox#1502] [Q&A]
Hello @beat-buesser, 
Thank you for your wonderful effort to ART. 

I have some issues to use HopskipJump attack with targeted label on CIFAR 10. Can you give me some example how to use it. While generating I am doing as follow: 

```
hopskipjump_attack = HopSkipJump(classifier=classifier,  verbose=False, targeted=True)   
hopskipjump_attack.generate(x=input_batch, y = np.array([0, 0, 0, 0, 0, 0, 0, 0, 0, 1]) )  
``` 
But it failed to succeed. 

Thank you beforehand!

**Comment by beat-buesser** [comment]
Hi @fukashi-hatake 

Thank you very much!

How does it fail? Is the attack not able to find an adversarial example or does the code return an error message?

**Reply by fukashi-hatake** [reply]
Thank you for your reply. 
The attack is not able to find an adversarial example even I increase the number of iterations.

**Reply by beat-buesser** [reply]
Is the label provided to `y` in method `generate` the true or target label? Does `classifier.predict(x=input_batch)`predict the correct class/true label?

**Reply by fukashi-hatake** [reply]
Is the label provided to y in the method generate the true or target label?
Answer: it is the target label 

Does classifier.predict(x=input_batch)predict the correct class/true label?
Answer: yes

**Reply by fukashi-hatake** [reply]
I just want to know that is it correct way to define target label? 
`hopskipjump_attack.generate(x=input_batch, y = np.array([0, 0, 0, 0, 0, 0, 0, 0, 0, 1]) )  `

**Reply by beat-buesser** [reply]
If `input_batch` contains a single sample it should work, to ensure correct shapes you could make `y` two-dimensional with `np.array([[0, 0, 0, 0, 0, 0, 0, 0, 0, 1]])`.

Are you using the default parameters of `HopSkipJump`?

**Reply by fukashi-hatake** [reply]
Are you using the default parameters of HopSkipJump?
Answer: yes (only targeted=True)

**Reply by beat-buesser** [reply]
It could be that the initialisation step of `HopSkipJump` is not able to sample a random starting perturbation that makes the sample adversarial. Have you tried increasing the argument `init_size` to allow for more samples in the initialisation step?

Is the attack successful in the untargeted case for this sample (with true label for `y` in generate)?
